src/ReadonlyRepository.ts

```diff
@@ -196,15 +196,15 @@ export class ReadonlyRepository<T extends Entity> implements IReadonlyRepository
                 }
 
                 const populateWhere = {
-                  [populateRepository.model.primaryKeyColumn.propertyName]: result[populate.propertyName] as EntityFieldValue,
+                  [populateRepository.model.primaryKeyColumn.propertyName]: result[populate.propertyName as keyof T] as EntityFieldValue,
                   ...populate.where,
                 };
 
                 populateQueries.push(
                   (async function populateModel(): Promise<void> {
                     const populateResult = await populateRepository.findOne({
-                      select: populate.select,
-                      where: populateWhere,
+                      select: populate.select as (string & keyof T)[],
```

Comment:
I'd like to avoid casting, if possible. This overrides the built in checking that typescript provides.

Reply:
I was able to make the PopulateArgs one a little prettier. I'll see if I can chip away at the others. Like I said in the comment below, it gets a little funky trying to get the typings to play nicely all the way down